### PR TITLE
Roaster unit models with updated docstring

### PR DESCRIPTION
**Pull request by jinliangma**
## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.

2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

**Comment by jinliangma** [comment]
The version of ree_oxalate_roaster.py contains the added Ca species, which needs to be merged.

File: src/prommis/roasting/ree_oxalate_roaster.py [hunk 1]
```diff
@@ -16,48 +16,119 @@
...
+IDAES REE Oxalate Roaster Unit Model
+====================================
+
+REE Oxalate Roaster is a unit operation to convert oxalates of rare earth and gangue metal
+elements to metal oxides through thermal decomposition and oxidation. There are 18 rare earth elements
+including `Sc`, `Y`, `La`, `Ce`, `Pr`, `Nd`, `Pm`, `Sm`, `Eu`, `Gd`, `Tb`, `Dy`, `Ho`, `Er`, `Tm`, `Yb`, `Lu`,
+and `Th` in the model. The user can specify a subset of those 18 elements through configuration when creating
+the unit model. The 3 gangue elements considered in the model are `Fe`, `Al`, and `Ca`.
+
+The feed oxalate mixture stream is typically from a precipiator in an upstream process. It is assumed
```
Comment (MarcusHolly) [inline src/prommis/roasting/ree_oxalate_roaster.py:29]:
```suggestion
The feed oxalate mixture stream is typically from a precipitator in an upstream process. It is assumed
```

File: src/prommis/roasting/ree_oxalate_roaster.py [hunk 2]
```diff
@@ -16,48 +16,119 @@
...
+that all oxalates are in their hydrate forms. In case the anhydrous oxalate feed flow rates are specified in
+the property package of the solid feed stream, the mole flow rates are converted to the corresponding hydrate flow rates.
```
Comment (MarcusHolly) [inline src/prommis/roasting/ree_oxalate_roaster.py:31]:
```suggestion
the property package of the solid feed stream, the molar flow rates are converted to the corresponding hydrate flow rates. 
```

File: src/prommis/roasting/ree_oxalate_roaster.py [hunk 3]
```diff
@@ -16,48 +16,119 @@
...
+
+The gas phase properties are calculated based on user configured property package.
+
+
+Mass Balance
+------------
+
+The content of the surface moisture in the solid feed stream is vaporized to enter the gas phase.
+
+The species mass balance is based on complete conversion of solid reactants such that the mole flow rates of
```
Comment (MarcusHolly) [inline src/prommis/roasting/ree_oxalate_roaster.py:93]:
```suggestion
The species mass balance is based on complete conversion of solid reactants such that the molar flow rates of
```

File: src/prommis/roasting/ree_oxalate_roaster.py [hunk 4]
```diff
@@ -16,48 +16,119 @@
...
+individual metals (rare earth and gaugue elements) are conserved. For the species in the gas phase, the :ce:`O2`
+is consumed while :ce:`CO2` and :ce:`H2O` are produced. For any other species in the gas feed stream that does not
+participate in any reactions, its mole flow rate in the gas product stream is the same as that in the
```
Comment (MarcusHolly) [inline src/prommis/roasting/ree_oxalate_roaster.py:96]:
```suggestion
participate in any reactions, its molar flow rate in the gas product stream is the same as that in the
```

File: src/prommis/roasting/ree_oxalate_roaster.py [hunk 5]
```diff
@@ -16,48 +16,119 @@
...
+Reactions
+---------
+
+The thermal decomposition reactions modeled for rare earth oxalates and three gaugue oxalates are as
+listed below:
+
+- :ce:`RE2(C2O4)3 \cdot xH2O(s) + 1.5O2(g) -> RE2O3(s) + 6CO2(g) + xH2O(g)`
+- :ce:`Fe2(C2O4)3 \cdot 2H2O(s) + 1.5O2(g) -> Fe2O3(s) + 6CO2(g) + 2H2O(g)`
+- :ce:`Al2(C2O4)3 \cdot H2O(s) + 1.5O2(g) -> Al2O3(s) + 6CO2(g) + H2O(g)`
+- :ce:`CaC2O4 \cdot H2O(s) + 0.5O2(g) -> CaO(s) + 2CO2(g) + H2O(g)`
```
Comment (MarcusHolly) [inline src/prommis/roasting/ree_oxalate_roaster.py:57]: cdot is not displaying correctly here (and elsewhere in the documentation)

**Review by MarcusHolly — COMMENTED** [review]
Just one typo to address

File: src/prommis/roasting/ree_oxalate_roaster.py (hunk 5 above)
Comment (bpaul4) [reply src/prommis/roasting/ree_oxalate_roaster.py:57]: This is also my only suggested change, might need a double slash `\\cdot`.

Comment (MarcusHolly) [reply src/prommis/roasting/ree_oxalate_roaster.py]: It's still not displaying correctly. I'm not sure of the proper way of doing this w/o using :math:, but I believe you can just google centered dot and paste that symbol instead (·).

**Review by MarcusHolly — APPROVED** [review]
LGTM

**Review by bpaul4 — APPROVED** [review]

**Pull request merged**